src/components/application_manager/src/commands/hmi/sdl_activate_app_request.cc

```diff
@@ -38,6 +38,46 @@ namespace application_manager {
 
 namespace commands {
 
+namespace {
+struct ProtoV4AppsOnDevice : std::unary_function<ApplicationSharedPtr, bool> {
+  connection_handler::DeviceHandle handle_;
+  ProtoV4AppsOnDevice(const connection_handler::DeviceHandle handle)
+      : handle_(handle) {}
+  bool operator()(const ApplicationSharedPtr app) const {
+    return app
+               ? handle_ == app->device() &&
+                     ProtocolVersion::kV4 == app->protocol_version()
+               : false;
+  }
+};
+
+struct ForegroundApp
+    : std::unary_function<SDLActivateAppRequest::V4ProtoApps::value_type,
+                          bool> {
+  bool operator()(
+      const SDLActivateAppRequest::V4ProtoApps::value_type ptr) const {
+    return ptr.valid() ? ptr->is_foreground() : false;
```

Comment:
shared pointer has overloaded `opeartor bool()` so you don't need to explicitly call 'valid'